case-lib/lib.sh

```diff
@@ -680,13 +680,26 @@ is_zephyr()
     test "$znum" -gt 10
 }
 
+get_ipc4_fw_name()
+{
+    local platf; platf=$(sof-dump-status.py -p)
+    case "$platf" in
+        tgl|adl)
```

Comment:
that will break sof-test on CML/SKL and not work for RPL either. 
don't we have a shortcut for cAVS platforms?

Reply:
@marc-hb @plbossart  it is not reliable to extract the firmware and topology used from kernel log, because it fails sometimes due to limited dmesg buffer.

Better fixes should be:
- CI records the what has been deployed and passes the information to sof-test.
- The kernel exposes what it has loaded(firmware/topology) to sysfs/proc.

comments?

Reply:
> because it fails sometimes due to limited dmesg buffer.

I don't like parsing logs but `dmesg` is never a good excuse because there is simply never any reason to use `dmesg` in any script for the exact reason you just mentioned - and others. `journalctl -k` should always be used instead, it's also better for other reasons. Let's treat `dmesg` like a dirty word: it can be used in private but never mentioned in public :-)

Of course you already know that because you... did all the `journalctl` work!!
- https://github.com/thesofproject/sof-test/pull/639

Reply:
> CI records what has been deployed and passes the information to sof-test.

If there is some bug or configuration issue in CI (and with IPC4 there has been VERY MANY!) then this will NOT match what the kernel actually loaded.

> The kernel exposes what it has loaded(firmware/topology) to sysfs/proc, https://github.com/thesofproject/linux/issues/3867

Either 3867 or parsing debug logs, looking for the last firmware that was loaded. We already (try to) do that in `verify-sof-firmware-load.sh`

Reply:
For tigerlake ipc4, sof-tgl.ri is renamed to dsp_basefw.bin, so the version major number can not be used to determine firmware name solely.